### PR TITLE
Dynk Updates - Part II

### PR DESCRIPTION
**Pull request by vkbo**
This PR finishes the rewrite of parseFun and some of the related routines.
This should be merged after #608 as that branch was merged into this due to the new tests added there.

~~There's still a Part III to come that finishes the rest, but that is mostly cleanup and error message formatting. This one completes all the important stuff.~~

Edit: The rest of the module has now been updated too, so this is the end of this.

### CODE REVIEW

File: source/dynk.f90 [hunk 1]
```diff
@@ -529,1080 +512,873 @@ subroutine dynk_parseFUN(gFields, lFields, nFields, inLine, iErr)
...
-        ! That program should then send a value back (in ASCII), which will be the new setting.
+  case("PIPE")
+    ! PIPE: Use a pair of UNIX FIFOs.
+    ! Another program is expected to hook onto the other end of the pipe,
+    ! and will recieve a message when SixTrack's dynk_computeFUN() is called.
+    ! That program should then send a value back (in ASCII), which will be the new setting.
 
-        call dynk_checkargs(nFields,7,"FUN funname PIPE inPipeName outPipeName ID fileUnit" )
-        call dynk_checkspace(1,0,4)
+    call dynk_checkargs(nSplit,6,"FUN funname PIPE inPipeName outPipeName ID fileUnit" )
```
Comment (kyrsjo) [inline source/dynk.f90]: Error message still says fileUnit!

**Comment by vkbo** [comment]
Fixed your comment, and sure, let's add a test.

**Comment by kyrsjo** [comment]
Great! Let's merge once the FIR and IIR tests are in. It would be great to have a PIPE test also, however it may not be so easy to integrate with the testing framework...

**Comment by kyrsjo** [comment]
The tests should go in master first, then merged here, of course...

**Comment by vkbo** [comment]
Yes, let's add them like I did the other tests. Make the canonicals on master, and merge them here to verify that they still pass.

Also not tested is ONOFF and PELP.

**Comment by kyrsjo** [comment]
IIR test in PR #618

**Comment by kyrsjo** [comment]
And you added an ONOFF test, right?

**Pull request merged**